From: Carlos Garnacho <carlosg@gnome.org>
Date: Fri, 5 Jun 2020 18:31:55 +0200
Subject: clutter/main: Drop dead code branches

We shouldn't get an input event that has not a device.

https://gitlab.gnome.org/GNOME/mutter/-/merge_requests/1486
---
 clutter/clutter/clutter-main.c | 40 ++++++----------------------------------
 1 file changed, 6 insertions(+), 34 deletions(-)

diff --git a/clutter/clutter/clutter-main.c b/clutter/clutter/clutter-main.c
index 2f6f4d1..6162db2 100644
--- a/clutter/clutter/clutter-main.c
+++ b/clutter/clutter/clutter-main.c
@@ -1774,25 +1774,9 @@ _clutter_process_event_details (ClutterActor        *stage,
                   break;
                 }
 
-              /* if the backend provides a device then we should
-               * already have everything we need to update it and
-               * get the actor underneath
-               */
-              if (device != NULL)
-                {
-                  actor = clutter_input_device_update (device, NULL,
-                                                       CLUTTER_STAGE (stage),
-                                                       TRUE);
-                }
-              else
-                {
-                  CLUTTER_NOTE (EVENT, "No device found: picking");
-
-                  actor = _clutter_stage_do_pick (CLUTTER_STAGE (stage),
-                                                  x, y,
-                                                  CLUTTER_PICK_REACTIVE);
-                }
-
+              actor = clutter_input_device_update (device, NULL,
+                                                   CLUTTER_STAGE (stage),
+                                                   TRUE);
               if (actor == NULL)
                 break;
 
@@ -1900,21 +1884,9 @@ _clutter_process_event_details (ClutterActor        *stage,
                   break;
                 }
 
-              if (device != NULL)
-                {
-                  actor = clutter_input_device_update (device, sequence,
-                                                       CLUTTER_STAGE (stage),
-                                                       TRUE);
-                }
-              else
-                {
-                  CLUTTER_NOTE (EVENT, "No device found: picking");
-
-                  actor = _clutter_stage_do_pick (CLUTTER_STAGE (stage),
-                                                  x, y,
-                                                  CLUTTER_PICK_REACTIVE);
-                }
-
+              actor = clutter_input_device_update (device, sequence,
+                                                   CLUTTER_STAGE (stage),
+                                                   TRUE);
               if (actor == NULL)
                 break;
 
